File: 백준/Silver/5555. 반지/반지.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    static int N, answer;
    static String word;

    static int stoi(String s) {
        return Integer.parseInt(s);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        word = br.readLine();
        N = stoi(br.readLine());

        for (int i = 0; i < N; i++) {
            String ring = br.readLine();
            ring = ring + ring.substring(0, ring.length() - 1);

            if (ring.contains(word)) {
                answer++;
            }
        }

        System.out.println(answer);
    }
}

File: 백준/Silver/5555. 반지/반지.kt
fun main() = with(System.`in`.bufferedReader()) {
    var count = 0
    val word = readLine()

    repeat(readLine().toInt()) {
        val input = readLine()
        val ring = input + input.substring(0, input.length - 1)

        if (ring.contains(word)) {
            count++
        }
    }

    print(count)
}